BLUESPAWN-client/src/mitigation/mitigations/MitigateV1093.cpp

```diff
@@ -19,29 +20,29 @@ namespace Mitigations {
 		) {}
 
 	bool MitigateV1093::MitigationIsEnforced(SecurityLevel level) {
-		LOG_INFO("Checking for presence of " << name);
-		
-		auto key = RegistryKey{ HKEY_LOCAL_MACHINE, L"System\\CurrentControlSet\\Control\\Lsa", L"RestrictAnonymous" };
-		if (!key.ValueExists()) {
-			return false;
+		std::map<RegistryKey, std::vector<RegistryValue>> keys;
+		auto key = RegistryKey{ HKEY_LOCAL_MACHINE, L"System\\CurrentControlSet\\Control\\Lsa" };
+
+		keys.emplace(key, CheckValues(key, {
+			{ L"RestrictAnonymous", RegistryType::REG_DWORD_T, 1, true, CheckDwordEqual },
+		}));
+
+		for (const auto& key : keys) {
+			for (const auto& value : key.second) {
+				LOG_INFO(L"[" + name + L"] RestrictAnonymous value is not set to 1");
+				return false;
+			}
 		}
 
-		if (key.Get<DWORD>() != 1) {
-			LOG_INFO("[V-1093 - Anonymous enumeration of shares must be restricted] RestrictAnonymous value is not set to 1");
-			return false;
-		}
-		
 		return true;
 	}
 
 	bool MitigateV1093::EnforceMitigation(SecurityLevel level) {
-		auto key = RegistryKey{ HKEY_LOCAL_MACHINE, L"System\\CurrentControlSet\\Control\\Lsa", L"RestrictAnonymous" };
-		if (!key.ValueExists()) {
-			DWORD value = 1;
-			return key.Create(&value, 4, REG_DWORD);
-		}
+		auto key = RegistryKey{ HKEY_LOCAL_MACHINE, L"System\\CurrentControlSet\\Control\\Lsa" };
+		std::wstring value = L"restrictanonymous";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'value' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=4967854)